lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,41 @@
+require 'faraday'
+require 'json'
+
+def connection
+  Faraday.new(
+    url: 'http://localhost:4000/api/tweets',
+    params: {},
+    headers: {
+      'Accept' => 'application/json',
+      'Content-Type' => 'application/json'
+    }
+  )
+end
+
+def tweets
+  response = connection.get
+
+  JSON.parse(response.body)['data'].map do |tweet|
+    {
+      author: tweet['author'],
+      id: tweet['id'],
+      message: tweet['message']
+    }
+  end
+end
+
+def parse_tweets(author)
+  tweets.select { |tweet| tweet[:author].downcase == author.downcase }
+end
+
+def create_tweet(author, message)
+  connection.post { |post| post.body = { "tweet": { "author": author, "message": message } }.to_json }
```

Comment:
I think the request to create a tweet may give us some relevant data in the response. Probably the data of created tweet would be there. You can parse it into a hash and return from this method, so someone could use it like this:

```ruby
created_tweet = create_tweet("Jon", "Hey!")
puts created_tweet[:id]
puts created_tweet[:message]
```